### PR TITLE
build: run unit tests with partial compilation and Angular linker

### PR DESCRIPTION
**Pull request by devversion**
Instead of running e2e tests against a linker-processed application, we
run all of our unit tests using partial compilation and the Angular
linker. This ensures we test our components more throroughly against
the Angular linker & partial compilation. This also solves our CI
flakiness issue due to the large amount of webdriver e2e tests running
concurrently on CI (especially since all e2e tests ran on snapshot
builds).

### CODE REVIEW

**Review by crisbeto — APPROVED** [review]
LGTM, although I don't have a lot of context on the Bazel code.

File: tools/linker-process/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    srcs = glob(["**/*.ts"]),
+    deps = [
+        "@npm//@angular/compiler-cli",
+        "@npm//@babel/core",
+        "@npm//@babel/traverse",
+        "@npm//@types/node",
+    ],
+)
+
+nodejs_binary(
```
Comment (jelbourn) [inline tools/linker-process/BUILD.bazel:17]: Could you add a comment explaining the purpose of this target?

File: tools/linker-process/index.bzl [hunk 2]
```diff
@@ -0,0 +1,92 @@
+load("@build_bazel_rules_nodejs//:providers.bzl", "JSModuleInfo", "JSNamedModuleInfo")
+
+"""Gets the Bazel manifest path for a given file."""
```
Comment (jelbourn) [inline tools/linker-process/index.bzl:3]: Would it make sense to elaborate somewhere in this file what a manifest is?

File: tools/linker-process/index.bzl [hunk 3]
```diff
@@ -0,0 +1,92 @@
+load("@build_bazel_rules_nodejs//:providers.bzl", "JSModuleInfo", "JSNamedModuleInfo")
```
Comment (jelbourn) [inline tools/linker-process/index.bzl:1]: Could we add a brief overview of what this file is for?

**Review by jelbourn — APPROVED** [review]
lgtm

File: tools/linker-process/BUILD.bazel (hunk 1 above)
Comment (devversion) [reply tools/linker-process/BUILD.bazel:17]: 👍

File: tools/linker-process/index.bzl (hunk 2 above)
Comment (devversion) [reply tools/linker-process/index.bzl:3]: I've added a little example of how these paths look like and where they usually reside in Bazel.

File: tools/linker-process/index.bzl (hunk 3 above)
Comment (devversion) [reply tools/linker-process/index.bzl:1]: Done.

**Review by josephperrott — APPROVED** [review]
LGTM

**Pull request merged**